chain/state/statetree.go

```diff
@@ -253,21 +251,17 @@ func (st *StateTree) Flush(ctx context.Context) (cid.Cid, error) {
 
 	for addr, sto := range st.snaps.layers[0].actors {
 		if sto.Delete {
-			if err := st.root.Delete(ctx, string(addr.Bytes())); err != nil {
+			if err := st.root.Delete(adt.AddrKey(addr)); err != nil {
 				return cid.Undef, err
 			}
 		} else {
-			if err := st.root.Set(ctx, string(addr.Bytes()), &sto.Act); err != nil {
+			if err := st.root.Put(adt.AddrKey(addr), &sto.Act); err != nil {
 				return cid.Undef, err
 			}
 		}
 	}
 
-	if err := st.root.Flush(ctx); err != nil {
-		return cid.Undef, err
-	}
-
-	return st.Store.Put(ctx, st.root)
+	return st.root.Root()
```

Comment:
We're dropping the context here, but we're not exactly consistent with our use of contexts anyways. Is this an issue?